syntax = "proto3";
import "request.proto";

service user {
  // login
  rpc login(app.v1.Request) returns (app.v1.Response) {}
  rpc listUser(app.v1.Request) returns (app.v1.Response) {}

  rpc register(app.v1.Request) returns (app.v1.Response) {}

  rpc delete(app.v1.Request) returns (app.v1.Response) {}

  rpc resetPassword(app.v1.Request) returns (app.v1.Response) {}
  rpc update(app.v1.Request) returns (app.v1.Response) {}
  rpc generatePassword(app.v1.Request) returns (app.v1.Response) {}
  rpc checkToken(app.v1.Request) returns (app.v1.Response) {}
  rpc query(app.v1.Request) returns (app.v1.Response) {}
}